### ISSUE TITLE
Feture Request: need to turn off auto cluster connection

### ISSUE THREAD

**Issue by Revolution1** [vscode-kubernetes-tools/vscode-kubernetes-tools#1191]
I have some clusters that need extra sso login to authenticate. And all of them is managed in the same kubeconfig with my local kind cluster.
The kubeconfig is configured to auto popup a webpage to let me login.

I only want to access the cluster when I need, but I cannot disable the kuberneted plugin cluster management feature. So everytime I open vscode, or the session expires, several webpage will popup, which is very annoying.

Better to have a setting to turn off auto connection or have a filter to let me choose which contexts I need to enable auto connection.

**Comment by Tatsinnit** [comment]
💡 Thank you so much @Revolution1 for observation share, I see. We could actually make it `enable/disable` based on the user choice variable, the whole dance of auth is happening due to AAD Cluster auth at the back ground which you have kindly pointed out already.

The above scenario happens when auto auth kicks in when the cluster in current-context is AAD-enabled cluster.

Essentially what I am proposing here is to add a user settings: `disable-add-cluster-auth` settings and then if set to `true` we skip the whole `auto-auth` mechanism.

On that, let me expand this for the idea with @peterbom for thoughts on above and auth set-up and this idea along with fyi to @lstocchi , @squillace and @itowlson ❤️ 🙏 Thanks all.

**Comment by squillace** [comment]
Yes, if we do most anything "automagically" that has ux consequences we need to have a setting to turn it off. I still dislike many popups I get, and one day I'm going to go through every single extension I use and destroy them all in settings. :-)

Please add the setting, and assume that any new "automatic" setting needs a way to turn it off in config.

**Comment by peterbom** [comment]
Hi @Revolution1 - a quick question before we press ahead with creating a new setting:

Is your SSO-authenticated cluster _actually_ an Azure AAD cluster that you merged into your `kubeconfig` using the AKS extension? That situation does indeed cause the situation you're describing, but you might have got there by a different route, in which case the setting that @Tatsinnit is proposing might not actually work.

Fundamentally I think there are a couple of ways to approach this:
1. Try to ensure that configs with interactive authentication never make it into the `kubeconfig`, which is what the setting above would achieve (imperfectly, because we can't control what users to with their `kubeconfig` files via the command line).
2. Load and read the `kubeconfig` in JS before passing it to `kubectl`. If it contains any interactive login, decide how to proceed (this could be based on a VS Code setting - `skip-interactive-login` or something).

Note that thoughtfully implementing (2) could also encompass [the proposed fix](https://github.com/vscode-kubernetes-tools/vscode-kubernetes-tools/issues/458#issuecomment-468874162) for another longstanding bug raised by @squillace.

**Comment by passcod** [comment]
I'm hitting this as well, with AWS SSO logins. It's to the point where I disable the extension entirely unless I really need it, because otherwise I can get five prompts in the same minute.

**Comment by wesbrownfavor** [comment]
Is there any update on this?
